lib/Delivery.js

```diff
@@ -275,10 +274,22 @@ class Delivery extends EventEmitter {
           this.courier.update(data.courier);
         }
 
-        this.status = data.status;
+        if (!this.pickup) {
+          this.pickup = new Pickup(data.pickup);
+        }
         this.pickup.eta = data.pickup.eta;
+
+        if (!this.dropoff) {
+          this.dropoff = new Dropoff(data.dropoff);
+        }
         this.dropoff.eta = data.dropoff.eta;
 
+        const status = (data || {}).status || 'unknown';
+        this.status = data.status;
+
+        // Emit the status event before determining if the current status is terminal
+        this.emit('status', status);
```

Comment:
This is the most urgent fix - without moving this above the code to kill the event handlers we aren't able to receive the fact that the delivery has completed